docs/data/data-grid/column-menu/column-menu.md

```diff
@@ -0,0 +1,164 @@
+
+```tsx
+function CustomColumnMenu(props: GridColumnMenuProps) {
+  return (
+    <GridColumnMenuDefault
+      {...props}
+      components={{
+        // Override `ColumnMenuFilterItem` component
+        ColumnMenuFilterItem: CustomFilterItem,
+      }}
+    />
+  );
+}
+```
+
+### Hiding a menu item
+
+To hide a menu item, you can set its respective slot in `components` to `null`.
+Check [this table](/x/react-data-grid/column-menu/#column-menu-items-and-their-configurations) to know the slot name for each menu item.
+
+```tsx
+function CustomColumnMenu(props: GridColumnMenuProps) {
+  return (
+    <GridColumnMenuDefault
+      {...props}
+      components={{
+        // Hide `ColumnMenuColumnsItem`
+        ColumnMenuColumnsItem: null,
+      }}
+    />
+  );
+}
+```
+
+### Reordering menu items
+
+Every item is assigned a `displayOrder` based on which it is shown before or after other items. It works in ascending order; the smaller the number is, the earlier it is displayed on the list. For new items default value for `displayOrder` is **100**.
+
+You can override `displayOrder` for the default or new items in `componentsProps`.
+
+Check [this table](/x/react-data-grid/column-menu/#column-menu-items-and-their-configurations) to see default `displayOrder` for each menu item.
+
+```tsx
+function CustomColumnMenu(props: GridColumnMenuProps) {
+  return (
+    <GridColumnMenuDefault
+      {...props}
+      componentsProps={{
+        // Swap positions of filter and sort items
+        columnMenuFilterItem: {
+          displayOrder: 0, // Previously `10`
+        },
+        columnMenuSortItem: {
+          displayOrder: 10, // Previously `0`
+        },
+      }}
+    />
+  );
+}
+```
+
+The following demo implements adding, overriding, hiding, and re-ordering of column menu items.
+
+{{"demo": "ReuseColumnMenuGrid.js", "bg": "inline"}}
+
+### Column menu items and their configurations
+
+As a reference, here are the default `components` supported by each column menu design along with the default item components and `displayOrder`.
+
+{{"demo": "ColumnMenuGridReferences.js", "bg": "inline", "hideToolbar": true}}
+
+## Custom menu component
+
+You can also customize and replace the column menu by [passing a fully custom component](/x/react-data-grid/components/#overriding-components) to the `ColumnMenu` slot of the Data Grid. If you want to add some of the default menu items to your custom component, you can import and re-use them.
+
+{{"demo": "CustomColumnMenuGrid.js", "bg": "inline"}}
+
+:::info
+<strong>Tip</strong>: In the `ColumnMenu` component and its items, you receive the prop `colDef` corresponding to the current column; you can use this to conditionally render some items or change some logic.
+:::
+
+## Column menu with Pro/Premium options [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan)[<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan)
```

Comment:
I think that we should move Pro/Premium features h2 of the docs to the bottom of the page to not alienate the MIT community. Unless there is a strong reason to have it in the middle of the page.

Reply:
But I'm not sure what, as a developer, I learn from this demo, so maybe?

```suggestion
```

Reply:
Actually, I thought also about keeping `Pro` and `Premium` menu items part of respective feature pages, but as we have a dedicated page for column menu, I thought that seeing the full potential of the menu might be encouraging for the community users to upgrade 🙈

Reply:
Also, to make the customizations easy to use as we have to provide the default menu configurations for all the items (including Pro and Premium ones) which we show in [this table](https://deploy-preview-6619--material-ui-x.netlify.app/x/react-data-grid/column-menu/#column-menu-items-and-their-configurations), so may it's better if we also have a demo to relate to?

We can move it to the end though if we agree on keeping it.